main.rb

```diff
@@ -0,0 +1,115 @@
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    x = 0
+    while x < length
+      yield(self[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    result = []
+    my_each { |x| result << x if yield(x) }
+    result
+  end
+
+  def my_all?(*)
+    unless block_given?
+      my_each { |i| return false if i.nil? || i == false }
+      return true
+    end
+    count = 0
+    my_each { |i| count += 1 if yield(i) }
+    count == length
+  end
+
+  def my_any?(number = 0)
+    unless block_given?
+      if number.instance_of?(Class)
+        my_each { |x| return true if x.is_a? number }
+      elsif number.instance_of?(Regexp)
+        my_each { |x| return true if number.match?(x.to_s) }
+      elsif [nil, false].include?(number)
+        my_each { |x| return true if x == number }
+      end
+      return false
+    end
+    my_each { |x| return true if yield(x) }
+    false
+  end
```

Comment:
## my_any

- [ ] returns false if the block ever returns a value false or nil::range

```Ruby
false_block = |num| num > HIGHEST_VALUE } 
range.my_any?(&false_block)
```

_got NameError:
        undefined local variable or method `length' for 5..50:Range_

- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil

```Ruby
true_array = [nil, false, true, []]
true_array.my_any?  
```

_expected true
             got false_

- [ ] when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern

```Ruby
words[0] = 'cat'
words.my_any?('cat') 
```

_expected true
             got false_